oauth2client/contrib/gce.py

```diff
@@ -86,31 +95,57 @@ class AppAssertionCredentials(AssertionCredentials):
     """
 
     @util.positional(2)
-    def __init__(self, scope='', **kwargs):
+    def __init__(self, scope='', service_account_email='default', **kwargs):
         """Constructor for AppAssertionCredentials
 
         Args:
-            scope: string or iterable of strings, scope(s) of the credentials
-                   being requested. Using this argument will have no effect on
-                   the actual scopes for tokens requested. These scopes are
-                   set at VM instance creation time and won't change.
+            scope:
+                string or iterable of strings, scope(s) of the credentials
+                being requested. Using this argument will have no effect on
+                the actual scopes for tokens requested. These scopes are
+                set at VM instance creation time and won't change.
+            service_account_email:
+                the email for these credentials. This can be used with custom
+                service accounts, or left blank to use the default service
+                account for the instance. Usually the compute engine service
+                account.
         """
         if scope:
             warnings.warn(_SCOPES_WARNING)
-        # This is just provided for backwards compatibility, but is not
-        # used by this class.
-        self.scope = util.scopes_to_string(scope)
+
+        self._service_account_email = service_account_email
+        self._service_account_info = None
+        self._project_id = None
+
         self.kwargs = kwargs
 
         # Assertion type is no longer used, but still in the
         # parent class signature.
         super(AppAssertionCredentials, self).__init__(None)
-        self._service_account_email = None
+
+    @property
+    def scopes(self):
+        return self._retrieve_scopes(httplib2.Http().request)
+
+    @property
+    def service_account_email(self):
+        if not self._service_account_info:
+            self._service_account_info = _get_service_account_info(
+                self._service_account_email
+            )
+        return self._service_account_info['serviceAccount']
+
+    def _retrieve_scopes(self, http_request):
+        if not self._service_account_info:
+            self._service_account_info = _get_service_account_info(
+                self._service_account_email,
+                http_request=http_request
+            )
+        return self._service_account_info['scopes']
 
     @classmethod
     def from_json(cls, json_data):
-        data = json.loads(_from_bytes(json_data))
-        return AppAssertionCredentials(data['scope'])
+        return AppAssertionCredentials(email=json_data['email'])
```

Comment:
A scope argument doesn't actually do anything except throw a warning, so I thought I'd leave it out to avoid repeatedly throwing warnings on serialization/deserialization, thoughts?